src/main/java/org/jboss/netty/handler/ssl/SslHandler.java

```diff
@@ -1655,9 +1655,8 @@ public void run() {
 
                 if (futures != null) {
                     final ClosedChannelException cause = new ClosedChannelException();
-                    final int size = futures.size();
-                    for (int i = 0; i < size; i ++) {
-                        futures.get(i).setFailure(cause);
+                    for (ChannelFuture future : futures) {
```

Comment:
This implicitly creates an Iterator, which will generate garbage to be collected.